### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by Koc**
1. I've removed check for Symfony version because there is composer requirement for version 2.2 and check for 2.1 unnecessary https://github.com/sonata-project/SonataIntlBundle/blob/master/composer.json#L19
2. I've removed `SessionLocaleDetector` because this class is invalid: there is no method `getLocale` in session class and detector uses namespace instead of concrete class

### CODE REVIEW

**Comment by OskarStark** [comment]
@Koc can you please fix travis?

File: Timezone/UserBasedTimezoneDetector.php [hunk 1]
```diff
@@ -51,7 +51,5 @@ public function getTimezone()
         if ($user instanceof User) {
             return $user->getTimezone();
         }
-
-        return;
```
Comment (Soullivaneuh) [inline Timezone/UserBasedTimezoneDetector.php]: Why removing this?

Comment (Koc) [inline Timezone/UserBasedTimezoneDetector.php]: it is also dead code that do nothing

Comment (OskarStark) [inline Timezone/UserBasedTimezoneDetector.php]: but it changes the behaviour from return null to void

Comment (Koc) [inline Timezone/UserBasedTimezoneDetector.php]: Are you talking about https://wiki.php.net/rfc/void_return_type ?

Comment (OskarStark) [inline Timezone/UserBasedTimezoneDetector.php]: hmm i thought that ``return;`` will return ``null``, but if this is correct what you say, it is valid to remove this line

**Comment by Koc** [comment]
@OskarStark @Soullivaneuh https://github.com/sonata-project/SonataIntlBundle/blob/master/Locale/SessionDetector.php looks like class if invalid and can be removed. Because it depends on `Symfony\Component\HttpFoundation\Session` - which is namespace but not class. Also there is no method `getLocale` in this class. So can I just remove it?

File: Timezone/UserBasedTimezoneDetector.php (hunk 1 above)
Comment (Soullivaneuh) [inline Timezone/UserBasedTimezoneDetector.php]: > it is also dead code that do nothing

You can't say that. All another returns are inside `if` statements.

**Comment by Soullivaneuh** [comment]
@Koc You could give a try but this need to be BC.

**Comment by Koc** [comment]
@Soullivaneuh `SessionDetector` is invalid. Nobody can used it because it could not works).

**Comment by Koc** [comment]
The PR was updated, travis is green except 3.0 job (deps error).

**Comment by OskarStark** [comment]
any news in this @Koc can we get this PR green?

**Comment by Koc** [comment]
@OskarStark I've rebased the PR. If fails on travis on some branches but failure not related to this PR. Something wrong with dependencies.

**Comment by OskarStark** [comment]
Thank you @Koc

**Pull request merged**